release-lock-file-spec.md

```diff
@@ -90,117 +212,377 @@ Sources are declared in a key/value mapping.
-* `deployments`
-    * Required: No
-    * Type: Object(String: Object(String: *Deployment* object)
-    * Format: 
-        * All keys **must** be valid BIP122 URIs matching the regular expression `blockchain://[0-9a-fA-F]{64}/block/[0-9a-fA-F]{64}`
-        * All values **must** be objects which conform to the following format
-            * Type: Object(String: *Deployment* object)
-            * Format:
-                * All keys **must** be strings conforming to the regular expression `[_a-zA-Z][_a-zA-Z0-9]*`.
-                    * In the case that there is a single deployed instance this **should** use the same name as the contract.  
-                    * In the case that there are multiple deployed instances of the same contract a unique name should be used for each deployed instance.
-                * All values **must** be objects conforming to the *Deployment* object format (see below).
-
-
-#### The *Deployments* Object
-
-
-* `address`:
-    * Required: Yes
-    * Type: String
-    * Format: Hex encoded ethereum address of the deployed contract.
-* `deploy_transaction`:
-    * Required: No
-    * Type: String
-    * Format: [BIP122](https://github.com/bitcoin/bips/blob/master/bip-0122.mediawiki) URI which defines the transaction in which this contract was created.
-* `deploy_block`:
-    * Required: No
-    * Type: String
-    * Format: [BIP122](https://github.com/bitcoin/bips/blob/master/bip-0122.mediawiki) URI which defines the block in which this contract was created.
-* `link_dependencies`:
-    * Reqired: Required to have an entry for each link reference found in either the `bytecode` or `runtime_bytecode` fields.
-    * Type: Object
-    * Format:
-        * All keys **must** be strings which are formatted as valid link targets.
-        * All values **must** conform to *one of* the following formats:
-            * A hex encoded ethereum address.
-            * A [json pointer](https://tools.ietf.org/html/rfc6901) to another *Deployment* object in the release lock file.  The `address` from the resolved *Deployment* object should be used.
-            * An IPFS URI with a JSON point in the fragment portion of the URI.  The IPFS hash must resolves to a valid release lock file.  The json pointer **must** resolves to a *Deployment* object within the release lock file.  The `address` from the resolved *Deployment* object should be used.
-
-
-#### BIP122 URIs
+    * Keys **must** be valid package names matching the regular expression `[a-z][-a-z0-9]{0,213}`
+    * Values **must** be valid IPFS URIs which resolve to a valid *Release Lock File*
+
+
+## Object Definitions
+
+Definitions for different objects used within the release lockfile.  All
+objects allow custom fields to be included.  Custom fields **should** be
+prefixed with `x-` to prevent name collisions with future versions of the
+specification.
+
+
+### The *Package Meta* Object
+
+The *Package Meta* object is defined to have the following key/value pairs.
```

Comment:
++ on putting authors and license into the meta object...this is a good separation.